lib/src/rewrite.rs

```diff
@@ -197,14 +197,16 @@ impl<'repo> CommitRewriter<'repo> {
 
     /// If a merge commit would end up with one parent being an ancestor of the
     /// other, then filter out the ancestor.
-    pub fn simplify_ancestor_merge(&mut self) {
+    pub fn simplify_ancestor_merge(&mut self) -> Result<(), BackendError> {
         let head_set: HashSet<_> = self
             .mut_repo
             .index()
             .heads(&mut self.new_parents.iter())
+            .map_err(|err| err.expect_backend_error())?
```

Comment:
It seems better to let `simplify_ancestor_merge()` return the same error as `heads()` returns instead of mapping it here

Reply:
Done, thanks!